zcash_client_sqlite/src/wallet.rs

```diff
@@ -3221,9 +3221,38 @@ fn recipient_params<P: consensus::Parameters>(
     }
 }
 
+fn flag_previously_received_change(
+    conn: &rusqlite::Transaction,
+    tx_ref: TxRef,
+) -> Result<(), SqliteClientError> {
+    let flag_received_change = |table_prefix| {
+        conn.execute(
+            &format!(
+                "UPDATE {table_prefix}_received_notes 
+                 SET is_change = 1
+                 FROM sent_notes sn
+                 WHERE sn.tx = {table_prefix}_received_notes.tx
```

Comment:
What kind of join does this implicitly do? (both here and in the migration which has an identical query)

It might be useful to add a comment describing the intent of this query.

Reply:
This is implicitly an inner join, the same as you get whenever you use a `FROM` clause with multiple tables and then use `WHERE` to perform the joins.